src/errors/AbstractInstantiationError.js

```diff
@@ -0,0 +1,12 @@
+
+/**
+ * AbstractInstantiationError is thrown when an abstract class is instantiated directly.
+ * This is a workaround to the fact that ES6 does not formally support abstract classes.
+ */
+class AbstractInstantiationError extends Error {
+	constructor(className) {
```

Comment:
FWIW, `NotImplementedError` documented _its_ constructor with a proper JSDoc block. Consequently, I love `NotImplementedError` more.

Reply:
I think I had a reason for removing this one's documentation.

Reply:
UPDATE: it was because originally it didn't have a parameter originally.

Also I took out the useless "hurr durr this is a constructor" line